v2/pkg/engine/resolve/context.go

```diff
@@ -138,11 +139,26 @@ func (c *Context) SetRateLimiter(limiter RateLimiter) {
 }
 
 func (c *Context) SubgraphErrors() error {
-	return c.subgraphErrors
+	if len(c.subgraphErrors) == 0 {
+		return nil
+	}
+
+	// Ensure the errors are appended in an idempotent order
+	keys := make([]string, 0, len(c.subgraphErrors))
+	for k := range c.subgraphErrors {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var joined error
+	for _, k := range keys {
+		joined = errors.Join(joined, c.subgraphErrors[k])
+	}
+	return joined
```

Comment:
Should we put this functionality under the flag, as the joined output will slightly change?